clients/clients-integration-tests/src/test/java/org/apache/kafka/server/quota/CustomQuotaCallbackTest.java

```diff
@@ -69,10 +70,37 @@ public void testCustomQuotaCallbackWithControllerServer() throws InterruptedExce
                         && CustomQuotaCallback.COUNTERS.values().stream().allMatch(counter -> counter.get() > 0), 
                     "The CustomQuotaCallback not triggered in all controllers. "
             );
-            
+        
+        }
+    }
+
+    @ClusterTest(
+        types = {Type.CO_KRAFT, Type.KRAFT},
+        serverProperties = {
+            @ClusterConfigProperty(key = QuotaConfig.CLIENT_QUOTA_CALLBACK_CLASS_CONFIG, value = "org.apache.kafka.server.quota.CustomQuotaCallbackTest$MonitorableCustomQuotaCallback"),
         }
+    )
+    public void testMonitorableCustomQuotaCallbackWithCombinedMode(ClusterInstance cluster) {
+        assertEquals(2, MonitorableCustomQuotaCallback.metricNames.size());
+        MetricName metricName1 = MonitorableCustomQuotaCallback.metricNames.get(0);
```

Comment:
Instead of using static fields, I think it's better to retrieve the metrics from the brokers and controllers like I do in my test.